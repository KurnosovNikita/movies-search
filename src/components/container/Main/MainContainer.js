import React, {Fragment} from "react";
import CardsContainer from '../CardsContainer/CardsContainer';
import TextBlock from "../../presentational/TextBlock";
import InfoPanelContainer from '../InfoPanelContainer/InfoPanelContainer';

class MainContainer extends React.PureComponent {
    render() {
        const {movies, activeSortByLink, handleSortLinkClick, handleImageClick} = this.props;

        return (
          <main className="main">
              {movies.length ? (
                <Fragment>
                    <InfoPanelContainer moviesQuantity={movies.length}
                                        activeSortByLink={activeSortByLink}
                                        handleSortLinkClick={handleSortLinkClick}/>
                    < CardsContainer movies={movies}
                                     handleImageClick={handleImageClick}/>
                </Fragment>
              ) : (
                <TextBlock className="no-results" text='No films found'/>
              )}
          </main>
        );
    }
}

export default MainContainer;