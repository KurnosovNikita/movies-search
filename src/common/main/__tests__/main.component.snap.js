import React from 'react';
import {shallow} from 'enzyme';
import Main from '../main.component';

describe('Main component', () => {
    let props;
    beforeEach(() => {
        props = {
            movies: [],
            activeSortByLink: jest.fn(),
            handleSortLinkClick: jest.fn(),
            handleImageClick: jest.fn(),
        };
    });

    it('should render <Main> correctly', () => {
        const component = shallow((<Main {...props} />));

        expect(component).toMatchSnapshot();
    });
});
