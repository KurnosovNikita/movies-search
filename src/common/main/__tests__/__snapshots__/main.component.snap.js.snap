// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render <Main> correctly 1`] = `
<main
  className="main"
>
  <Text
    className="no-results"
    text="No  films found"
  />
</main>
`;
